devito/memory.py

```diff
@@ -8,10 +8,9 @@
 import numpy as np
 from sympy import Eq
 
-from devito.dimension import t
 from devito.logger import error
 from devito.tools import convert_dtype_to_ctype
-
+import devito
```

Comment:
`operator.py` imports `SymbolicData`, `CompositeData`, `Forward` and `Backward` from `interfaces.py`. I don't think we can remove all 4 of these dependencies. 

`interfaces.py` imports `CMemory` and `first_touch` from `memory.py`. These seem pretty obvious dependencies to me. 

If we do a `from devito import Operator` to initialise our own arrays (and why shouldn't we?), we introduce a dependency from `memory.py` to `operator.py` and hence completing the circle. 

I don't see any way to restructure this that would not introduce a circular dependency. Any ideas?